File: HoneyStoreUI/src/app/models/product.ts
import { Producer } from './producer';
import { Category } from './category';
import { ProductPhoto } from './product-photo';
import { DomSanitizer, SafeUrl } from '@angular/platform-browser';

export class Product {
    id: number;
    imageUrl: string | undefined;
    name: string;    
    mark: number;
    description: string | undefined;
    price: number;
    quantity: number;
    weight: number;
    commentsEnabled: boolean | undefined;
    producer: Producer;
    productPhotoId: number;
    productPhoto: ProductPhoto;
    categoryId: number | undefined;
    category: Category;

    constructor() {
        this.id = 0;
        this.name = '';   
        this.imageUrl = '';
        this.mark = 0;
        this.description = '';
        this.price = 10;
        this.quantity = 1;
        this.weight = 0;       
        this.commentsEnabled = true;
        this.productPhotoId = 0;
        this.productPhoto = new ProductPhoto();
        this.producer = new Producer();
        this.categoryId = undefined;
        this.category = new Category();
    }
}

File: HoneyStore.DataAccess/Entities/Product.cs
using HoneyStore.DataAccess.Interfaces;

namespace HoneyStore.DataAccess.Entities
{
    public class Product : IIdentifier
    {
        public int Id { get; set; }

        public string ImageUrl { get; set; }

        public string Name { get; set; }

        public decimal Price { get; set; }

        public string Description { get; set; }

        public int Mark { get; set; }

        public int Quantity { get; set; }

        public int Weight { get; set; }

        public bool CommentsEnabled { get; set; }

        public int ProducerId { get; set; }

        public Producer Producer { get; set; }

        public int? ProductPhotoId { get; set; }
        
        public ProductPhoto ProductPhoto { get; set; }

        public int? CategoryId { get; set; }

        public Category Category { get; set; }

        public ICollection<Comment> Comments { get; set; }
    }
}
